x-pack/plugins/case/server/common/models/commentable_case.ts

```diff
@@ -291,10 +275,33 @@ export class CommentableCase {
       },
     });
 
-    return CollectWithSubCaseResponseRt.encode({
+    const collectionTotalAlerts =
+      countAlertsForID({ comments: collectionComments, id: this.collection.id }) ?? 0;
+
+    const caseResponse = {
```

Comment:
The one downside to doing it this way is that if a comment is attached to a sub case, we will return all the comments for the sub case and the parent. That might be too much data 🤷

Reply:
We override parent's comments if we are on a subCase with the ones of the subCase.

It's:

```
...caseResponse,
comments: flattenCommentSavedObjects(subCaseComments.saved_objects),
```

Reply:
I can move the `getAllCaseComments` and `countAlertsForID` for the parent after the check for the subCase as an optimization.

Reply:
Christos and I talked about this, I'm cool with the way it is 👍